docs/source/developers/python.rst

```diff
@@ -55,12 +55,15 @@ like so:
 
 .. code-block:: shell
 
-   pytest pyarrow
+   pytest arrow/python/pyarrow
```

Comment:
This really depends on which directory your shell is currently in. If you just ran something like `pip install -e .`, then presumably you're already inside the `python` directory.

Also, I would expect Python programmers to understand how to get this right.

Reply:
That's certainly true, but _most_ of the other commands on this page are written assuming you are located just above the arrow repo. So for consistency (and for being more explicit), I think we can still update it.

And then any python programmer will still be able to understand how to edit it according to their location :)

(BTW, we actually ran into this because of the combination that the guide currently recommends to create a "pyarrow" virtualenv, and then doing `pytest pyarrow` following the other instructions, it tried to run the tests of the all packages in the virtualenv ... Now the virtualenv is renamed, that's of course less of an issue)

Reply:
I think the instructions should mimick the usual workflow, though. Is it common to run the PyArrow tests from the directory below the git checkout? I know I always work from the `python` directory.

Reply:
Yes, currently the instructions are written from the folder `repos`, which is above `arrow`. Even when building `pyarrow` you move from `repos` to `python` folder in `arrow` and than back to `repos` again.

I am sure there are standard ways of working but for someone already dealing with build and scale of Arrow library I think this simple things should be written explicitly and without adding more confusion.

What we can do is stay in python folder after building pyarrow (I would than change the instructions there) and leave pytest as it was?  What do u think?

Reply:
Hmm, I don't know, I never move back to the parent directory. Usually I would have one or several shells opened in the `cpp` directory and another one in the `python` directory, so I avoid back-and-forths.

So, yes, I would vote for staying in the `python` directory after building PyArrow.

Reply:
Personally, I always stay in the root arrow repo (and have aliases for things I regularly need that have the path baked in or do the pushd/popd). So that's already a different workflow :)

> What we can do is stay in python folder after building pyarrow (I would than change the instructions there) and leave pytest as it was? What do u think?

I would personally leave that for a separate PR to discuss (it would impact almost every command here)

Reply:
Ok.